### PR TITLE
EDU-3193: Enhance Role docs for clarity

### PR DESCRIPTION
**Pull request by fairlydurable**
Updates Role descriptions to better describe what each role can and cannot do

### CODE REVIEW

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 1]
```diff
@@ -105,18 +105,31 @@ When a Global Admin invites a user to join an account, the Global Admin selects
...
 - **Developer**
-  - Can create and update Namespaces; has full control over [Workflows](/workflows)
-  - Has Namespace Admin permissions for each Namespace created by that user
-- **Read-Only:** Can only read information
-
-In addition, there are two roles that the Global Admin cannot assign:
+  - Can create Namespaces and modify Namespaces they create
+  - Has Namespace Admin [permissions](#namespace-level-permissions) on all [Namespaces](/namespaces) they've created
+    - Has read-level access to the list of all users within the account so they can add and remove users from a Namespace they've created
+    - Can assign Namespace-level permissions (either read-only or write) to users for a Namespace they've created
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:111]: I believe they can also assign namespace admin permissions

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 2]
```diff
@@ -105,18 +105,31 @@ When a Global Admin invites a user to join an account, the Global Admin selects
...
+  - Has write permissions for all Namespaces they create
+    - Can control [Workflows](/workflows), [Schedules](/workflows#schedule), etc within those Namespaces
+- **Read-Only (non-administrative)**
+  - Must be assigned to a Namespace by a Namespace Admin
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:115]: Do you mean "assigned to a user"? Read-Only access for a namespace is assigned to a user

Comment (mattkim) [reply docs/production-deployment/cloud/account-setup/users.mdx:115]: Also this section is to describe the read-only account level role?

In that case they do not have access to any namespaces.

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 3]
```diff
@@ -105,18 +105,31 @@ When a Global Admin invites a user to join an account, the Global Admin selects
...
+  - By default, Read-Only users have read-level ("read-only") permissions within a [Namespace](/namespaces)
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:116]: I don't think this is correct either--read-only account role does not grant read permissions to namespaces

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 4]
```diff
@@ -105,18 +105,31 @@ When a Global Admin invites a user to join an account, the Global Admin selects
...
+  - Once added:
+    - Can view Namespace operational details, including [Workflows](/workflows), [Schedules](/workflows#schedule), etc
+    - Can view a list of users granted access to the same Namespace
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:119]: This is not correct either read-only cannot view namespace details nor can they get a list of users for the namespace

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 5]
```diff
@@ -105,18 +105,31 @@ When a Global Admin invites a user to join an account, the Global Admin selects
...
+    - Can't view user account details for other users within the same Namespace
+  - A Namespace Admin can [update the Namespace permissions of this role](#update-roles) to grant write permissions:
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:121]: a namespace admin can grant any permission. It is not limited to write permissions.

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 6]
```diff
@@ -105,18 +105,31 @@ When a Global Admin invites a user to join an account, the Global Admin selects
...
+    - Once updated, can create, rename, update, and delete Workflows, Schedules, etc within the Namespace
+    - The role name ("Read-Only") does not change even when granted write permissions for a Namespace
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:123]: I personally feel like specifying "account role" Read-Only makes this clearer.

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 7]
```diff
@@ -105,18 +105,31 @@ When a Global Admin invites a user to join an account, the Global Admin selects
...
+
+In addition, there are two roles that the Global Admin can't assign:
 
 - **Account Owner**
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:127]: I believe the account owner can also assign billing permissions? @captainbeardo to verify

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 8]
```diff
@@ -105,18 +105,31 @@ When a Global Admin invites a user to join an account, the Global Admin selects
...
-- **Read-Only:** Can only read information
-
-In addition, there are two roles that the Global Admin cannot assign:
+  - Can create Namespaces and modify Namespaces they create
+  - Has Namespace Admin [permissions](#namespace-level-permissions) on all [Namespaces](/namespaces) they've created
+    - Has read-level access to the list of all users within the account so they can add and remove users from a Namespace they've created
+    - Can assign Namespace-level permissions (either read-only or write) to users for a Namespace they've created
+  - Has write permissions for all Namespaces they create
+    - Can control [Workflows](/workflows), [Schedules](/workflows#schedule), etc within those Namespaces
+- **Read-Only (non-administrative)**
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:114]: In general I feel like there's not much to say about the Read-Only account role.

It basically is the "default" role that gives you access to the account.

I believe it allows you to view your own user profile page.

It does not give you any access to namespaces or even list users in the account (in the UI)

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 9]
```diff
@@ -259,59 +272,59 @@ Account Global Admin has Namespace Admin permissions on Namespaces.
...
+| GetRegions                            | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetRequestStatus                      | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetRequestStatuses                    |                    |           |               | ✔            | ✔             |
+| GetRequestStatusesForNamespace        | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetRequestStatusesForUser             | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetRoles                              | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetRolesByPermissions                 | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetServiceAccount                     | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetServiceAccounts                    | ✔                  | ✔         | ✔             | ✔            | ✔             |
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:309]: There are caveats with apis like this that we should eventually try to add documentation for.

For example I believe this api is part of the Read-Only Role but we have extra checks to ensure only namespace admins can manage namespace scoped service accounts

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 10]
```diff
@@ -259,59 +272,59 @@ Account Global Admin has Namespace Admin permissions on Namespaces.
...
+| GetStripeInvoice                      |                    |           | ✔             |              | ✔             |
+| GetUser                               | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetUsers                              | ✔                  | ✔         | ✔             | ✔            | ✔             |
+| GetUsersWithAccountRoles              | ✔                  | ✔         | ✔             | ✔            | ✔             |
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:313]: These apis are technically part of Read-Only role today but we are in the process of migrating to Admin role--we should leave this alone for now and not mention that in the docs.

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 11]
```diff
@@ -259,59 +272,59 @@ Account Global Admin has Namespace Admin permissions on Namespaces.
...
+| InviteUsers                           |                    |           |               | ✔            | ✔             |
+| ListCreditLedgerEntries               |                    |           | ✔             |              | ✔             |
+| ListGrants                            |                    |           | ✔             |              | ✔             |
+| ListMetronomeInvoices                 |                    |           | ✔             |              | ✔             |
+| ListMetronomeInvoicesForNamespace     |                    |           | ✔             |              | ✔             |
+| ListNamespaces                        | ✔                  | ✔         | ✔             | ✔            | ✔             |
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:319]: The caveat here is that it only returns the namespaces you have access to

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 12]
```diff
@@ -259,59 +272,59 @@ Account Global Admin has Namespace Admin permissions on Namespaces.
...
+| ListPromotionGrantBalances            |                    |           | ✔             |              | ✔             |
+| ResendUserInvite                      |                    |           |               | ✔            | ✔             |
+| SetAccountSettings                    |                    |           |               | ✔            | ✔             |
+| SyncCurrentUserInvite                 | ✔                  | ✔         | ✔             | ✔            | ✔             |
```
Comment (mattkim) [inline docs/production-deployment/cloud/account-setup/users.mdx:323]: We don't really use this api for anything I don't think it's even part of the public API?

Also just thinking this through--for the API table here should we only be listing the publicly available cloud ops apis?

https://github.com/temporalio/api-cloud

Comment (mattkim) [reply docs/production-deployment/cloud/account-setup/users.mdx:323]: https://github.com/temporalio/api-cloud/blob/main/temporal/api/cloud/cloudservice/v1/service.proto

**Review by mattkim — COMMENTED** [review]
on line 168:

> - **Write:** Can create, rename, update, and delete Workflows within the Namespace

I think we should remove the "rename" part here

File: docs/production-deployment/cloud/account-setup/users.mdx (hunk 1 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/account-setup/users.mdx:111]: Adding

**Review by mattkim — COMMENTED** [review]
line 174

> This does not apply to the Account Owner role.

this confuses me a little bit--need to check with @captainbeardo here on who can update the account owner role--can it be another account owner?

File: docs/production-deployment/cloud/account-setup/users.mdx (hunk 1 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/account-setup/users.mdx:111]:
```
  - Has Namespace Admin [permissions](#namespace-level-permissions) on all [Namespaces](/namespaces) they've created
    - Has read-level access to the list of all users within the account so they can add and remove users from a Namespace they've created
    - Can assign Namespace-level permissions (either read-only or write) to users for a Namespace they've created
    - Can assign Namespace Admin permissions to users for a Namespace they've created
```

File: docs/production-deployment/cloud/account-setup/users.mdx (hunk 2 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/account-setup/users.mdx:115]: Going to discuss on-thread what this means.

**Comment by fairlydurable** [comment]
Closing, renaming branch, and migrating to new fresh PR

**Pull request closed without merging**